app/controllers/switch-psp/switch-tasks.service.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+const { CREDENTIAL_STATE } = require('../../utils/credentials')
+
+const defaultTrue = () => true
+
+const credentialsComplete = function credentialsComplete (targetCredential, account) {
+  return [ CREDENTIAL_STATE.ENTERED, CREDENTIAL_STATE.VERIFIED, CREDENTIAL_STATE.ACTIVE, CREDENTIAL_STATE.RETIRED ]
+    .includes(targetCredential.state)
+}
+
+const worldpayTaskList = [{
+  key: 'LINK_CREDENTIALS',
+  isEnabled: defaultTrue,
+  isComplete: credentialsComplete
+}]
+
+const providerTaskListMap = {
+  worldpay: worldpayTaskList
+}
+
+function getStatusesFor (targetCredential, account) {
+  const schema = providerTaskListMap[targetCredential.payment_provider]
+  const taskList = schema.reduce((aggregate, taskItem) => {
+    aggregate[taskItem.key] = {
+      enabled: taskItem.isEnabled(targetCredential, account),
+      complete: taskItem.isComplete(targetCredential, account)
```

Comment:
I get the intention behind this, but it is a tiny bit confusing to follow.
Maybe you could talk me through what benefits are vs the following:

```
if (target.Credential.payment_provider === 'worldpay') {
    return {
        'LINK_CREDENTIALS': {
            enabled: true,
            complete: credentialsComplete(targetCredential, account)
        },
        ....
    }
}
if (target.Credential.payment_provider === 'stripe') {
   return {
       ....
   }
}
````